models/balancer/polygon/balancer_v2_polygon_trades.sql

```diff
@@ -0,0 +1,138 @@
+        swap.evt_tx_hash AS tx_hash,
+        '' AS trace_address,
+        swap.evt_index
+    FROM
+        swap_fees
+        INNER JOIN {{ source ('balancer_v2_polygon', 'Vault_evt_Swap') }} swap
+            ON swap.evt_block_number = swap_fees.evt_block_number
+            AND swap.evt_tx_hash = swap_fees.evt_tx_hash
+            AND swap.evt_index = swap_fees.evt_index
+        LEFT JOIN {{ ref('balancer_v2_polygon_pools_fees') }}
+            ON pools_fees.contract_address = swap_fees.contract_address
+            AND pools_fees.block_time = swap_fees.max_fee_evt_block_time
+    WHERE
+        swap.tokenIn != swap_fees.contract_address
+        AND swap.tokenOut != swap_fees.contract_address
+        -- Incremental logic.
+        {% if is_incremental() %}
+        AND swap.evt_block_time >= DATE_TRUNC("day", NOW() - interval '1 week')
+        {% endif %}
+)
+SELECT
+    'polygon' AS blockchain,
+    'balancer' AS project,
+    '2' AS version,
+    TRY_CAST (DATE_TRUNC('DAY', dexs.block_time) AS DATE) AS block_date,
+    dexs.block_time,
+    erc20a.symbol AS token_bought_symbol,
+    erc20b.symbol AS token_sold_symbol,
+    CASE WHEN LOWER(erc20a.symbol) > LOWER(erc20b.symbol) THEN
+        CONCAT(erc20b.symbol, '-', erc20a.symbol)
+    ELSE
+        CONCAT(erc20a.symbol, '-', erc20b.symbol)
+    END AS token_pair,
+    dexs.token_bought_amount_raw / POWER(10, erc20a.decimals) AS token_bought_amount,
+    dexs.token_sold_amount_raw / POWER(10, erc20b.decimals) AS token_sold_amount,
+    dexs.token_bought_amount_raw,
+    dexs.token_sold_amount_raw,
+    COALESCE(
+        dexs.amount_usd,
+        dexs.token_bought_amount_raw / POWER(10, p_bought.decimals) * p_bought.price,
+        dexs.token_sold_amount_raw / POWER(10, p_sold.decimals) * p_sold.price
+    ) AS amount_usd,
+    dexs.token_bought_address,
+    dexs.token_sold_address,
+    -- Subqueries rely on this COALESCE to avoid redundant joins with the transactions table.
+    COALESCE(dexs.taker, tx.from) AS taker,
+    dexs.maker,
+    dexs.project_contract_address,
+    dexs.swap_fee,
+    dexs.tx_hash,
+    tx.from AS tx_from,
+    tx.to AS tx_to,
+    dexs.trace_address,
+    dexs.evt_index
+FROM
+    dexs
+    INNER JOIN {{ source ('polygon', 'transactions') }} tx
+        ON tx.hash = dexs.tx_hash
+        -- Incremental logic.
+        {% if not is_incremental () %}
+        AND tx.block_time >= '{{ project_start_date }}'
+        {% endif %}
+        {% if is_incremental () %}
+        AND tx.block_time >= DATE_TRUNC("day", NOW() - interval '1 week')
+        {% endif %}
+    LEFT JOIN {{ ref ('tokens_erc20') }} erc20a
+        ON erc20a.contract_address = dexs.token_bought_address
+        AND erc20a.blockchain = 'polygon'
+    LEFT JOIN {{ ref ('tokens_erc20') }} erc20b
+        ON erc20b.contract_address = dexs.token_sold_address
+        AND erc20b.blockchain = 'polygon'
+    LEFT JOIN {{ source ('prices', 'usd') }} p_bought
```

Comment:
I will comment only on this model, but you have to `LEFT JOIN` with `view_bpt_prices` in every network. 

- https://github.com/balancer-labs/dune-spellbook/blob/main/deprecated-dune-v1-abstractions/ethereum/dex/trades/insert_balancer.sql#L121 
- https://github.com/balancer-labs/dune-spellbook/blob/main/deprecated-dune-v1-abstractions/ethereum/dex/trades/insert_balancer.sql#L43